tests/ui/redundant_test_prefix.rs

```diff
@@ -0,0 +1,11 @@
+#![warn(clippy::redundant_test_prefix)]
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_redundant_name() {
```

Comment:
Our UI test support error annotations. Could you add them?

These should work:

```suggestion
    fn test_redundant_name() {
        //~^ ERROR: this test is prefixed redundantly with `test`
        //~| HELP: consider removing the redundant `test` prefix from the test name
        //~| NOTE: `-D clippy::redundant-test-prefix` implied by `-D warnings`
        //~| HELP: to override `-D warnings` add `#[allow(clippy::redundant_test_prefix)]`
```